### ISSUE TITLE
Authenticate seems to be called again with on_behalf_of

### ISSUE THREAD

**Issue by rjnienaber** [CurrencyCloud/currencycloud-python#1]
Here is the code I was using to reproduce
```python
beneficiary_params = {
        'beneficiary_entity_type': 'individual',
        'beneficiary_first_name': 'Dr.',
        'beneficiary_last_name': 'Who',
        'beneficiary_address': ['Address 42'],
        'beneficiary_city': 'The Citadel',
        'beneficiary_postcode': '42424',
        'beneficiary_country': 'GB',
        'bank_account_holder_name': 'Test User',
        'bank_country': 'GB',
        'currency': 'GBP',
        'name': 'Test User Nick',
        'account_number': '4185 4372',
        'routing_code_type_1': 'aba',
        'routing_code_value_1': '123456780',
        'routing_code_type_2': 'sort_code',
        'routing_code_value_2': '400730',
        'payment_types': ['priority', 'regular']
    }

with currencycloud.on_behalf_of('1f056099-f3a5-4d54-9bad-939e41b576ef'):
    beneficiary = currencycloud.Beneficiary.create(**beneficiary_params)
    beneficiary = currencycloud.Beneficiary.create(**beneficiary_params)
```

And this is the error I was receiving:

```python
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "currencycloud/actions/create.py", line 6, in create
    return cls(**cls.post('create', **params))
  File "currencycloud/resource.py", line 96, in post
    cls.encode_parameters(params))
  File "currencycloud/request_handler.py", line 56, in post
    **kargs)
  File "currencycloud/request_handler.py", line 81, in __retry_authenticate
    self.session.reauthenticate()
  File "currencycloud/session.py", line 112, in reauthenticate
    self.authenticate()
  File "currencycloud/session.py", line 103, in authenticate
    should_retry=False
  File "currencycloud/request_handler.py", line 56, in post
    **kargs)
  File "currencycloud/request_handler.py", line 89, in __retry_authenticate
    return response_handler.process()
  File "currencycloud/response_handler.py", line 32, in process
    self.__handle_failure()
  File "currencycloud/response_handler.py", line 44, in __handle_failure
    raise error(self.verb, self.route, self.params, self.response)
currencycloud.errors.api.BadRequestError: BadRequestError
```
```yaml
---
errors:
- code: invalid_extra_parameters
  field: base
  message: Invalid extra parameters:'{:on_behalf_of=>"1f056099-f3a5-4d54-9bad-939e41b576ef"}'
  params:
    parameters:
      on_behalf_of: 1f056099-f3a5-4d54-9bad-939e41b576ef
platform: python - 2.7.6 (default, Jun 22 2015, 17:58:13) - CPython
request:
  parameters:
    api_key: <api key>
    login_id: <login id>
    on_behalf_of: 1f056099-f3a5-4d54-9bad-939e41b576ef
  url: https://devapi.thecurrencycloud.com/v2/authenticate/api
  verb: post
response:
  date: Wed, 01 Jul 2015 16:17:19 GMT
  request_id: 2820718196152030795
  status_code: 400
```

**Comment by alexiob** [comment]
this is what's going on:

```python

# first request, authentication because the api has not yet done it
('RESPONSE', 200, <Response [200]>, 'https://devapi.thecurrencycloud.com/v2/authenticate/api', {'login_id': 'rjnienaber@gmail.com', 'api_key': 'ef0fd50fca1fb14c1fab3a8436b9ecb65f02f129fd87eafa45ded8ae257528f0'}, {'headers': {}})

# the call returns a 401 (not authenticated) using on_behalf_of

('RESPONSE', 401, <Response [401]>, 'https://devapi.thecurrencycloud.com/v2/beneficiaries/create', {'beneficiary_entity_type': 'individual', 'on_behalf_of': '1f056099-f3a5-4d54-9bad-939e41b576ef', 'bank_country': 'GB', 'name': 'Test User Nick', 'currency': 'GBP', 'payment_types[]': ['priority', 'regular'], 'beneficiary_first_name': 'Dr.', 'beneficiary_last_name': 'Who', 'beneficiary_city': 'The Citadel', 'routing_code_type_2': 'sort_code', 'routing_code_type_1': 'aba', 'beneficiary_country': 'GB', 'account_number': '4185 4372', 'beneficiary_postcode': '42424', 'bank_account_holder_name': 'Test User', 'routing_code_value_1': '123456780', 'routing_code_value_2': '400730', 'beneficiary_address': ['Address 42']}, {'headers': {'X-Auth-Token': u'ccb9a597d2a251620aaa382c1c77c7e6'}})

#as this is a 401 error, the lib tries to re-authenticate, but there is the on_behalf_of parameter that gives an error

('RESPONSE', 400, <Response [400]>, 'https://devapi.thecurrencycloud.com/v2/authenticate/api', {'login_id': 'rjnienaber@gmail.com', 'on_behalf_of': '1f056099-f3a5-4d54-9bad-939e41b576ef', 'api_key': 'ef0fd50fca1fb14c1fab3a8436b9ecb65f02f129fd87eafa45ded8ae257528f0'}, {'headers': {}})
```

What do you want me to do here?

**Comment by rjnienaber** [comment]
This seems to have been fixed with fbe6873.
